tests/test_smt.py

```diff
@@ -0,0 +1,79 @@
+from hypothesis import (
+        given,
+        settings,
+        strategies as st,
+    )
+
+from trie.smt import (
+    SparseMerkleTree,
+    SparseMerkleProof,
+    calc_root,
+    BLANK_NODE,
+)
+
+
+@given(
+    k=st.binary(min_size=1, max_size=32),
+    v=st.binary(min_size=1, max_size=32),
+)
+def test_simple_kv(k, v):
+    smt = SparseMerkleTree(key_size=len(k))
+
+    # Nothing has been added yet
+    assert not smt.exists(k)
+
+    # Now that something is added, it should be consistent
+    smt.set(k, v)
+    assert smt.get(k) == v
+    assert smt.root_hash == calc_root(k, v, smt.branch(k))
+
+    # If you delete it, it goes away
+    smt.delete(k)
+    assert not smt.exists(k)
+
+
+@given(
+    data=st.data(),
+    key_size=st.integers(min_value=1, max_value=32),
+)
+#@settings(deadline=None)  # This option is not available in hypothesis-3.7.0
```

Comment:
This like should just be removed.

Reply:
It found a pretty decent bug I wouldn't have found otherwise, but can remove if upgrading isn't worth hassle.

Reply:
My comment about removing it is that it commenting it out is equivalent to removing it but with the negative downside of leaving it in the codebase.  If we're leaving it in it should get a comment explaining why.  If we are no longer using hypothesis versions that support this option I don't see why we should leave this in...